File: app/src/main/java/br/com/popularmovies/data/interceptor/AuthorizationInterceptor.java
package br.com.popularmovies.data.interceptor;

import org.jetbrains.annotations.NotNull;

import java.io.IOException;

import br.com.popularmovies.BuildConfig;
import okhttp3.HttpUrl;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

import static br.com.popularmovies.data.Constants.API_KEY;

public class AuthorizationInterceptor implements Interceptor {

    @NotNull
    @Override
    public Response intercept(@NotNull Chain chain) throws IOException {
        Request original = chain.request();
        HttpUrl originalHttpUrl = original.url();

        HttpUrl url = originalHttpUrl.newBuilder()
                .addQueryParameter(API_KEY, BuildConfig.MdbApiKey)
                .build();

        Request.Builder requestBuilder = original.newBuilder()
                .url(url);

        Request request = requestBuilder.build();
        return chain.proceed(request);
    }
}

File: app/src/main/java/br/com/popularmovies/core/network/retrofit/interceptor/AuthorizationInterceptor.kt
package br.com.popularmovies.core.network.retrofit.interceptor

import br.com.popularmovies.BuildConfig
import br.com.popularmovies.core.network.API_KEY
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class AuthorizationInterceptor : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val original = chain.request()
        val originalHttpUrl = original.url

        val url = originalHttpUrl.newBuilder()
            .addQueryParameter(API_KEY, BuildConfig.MdbApiKey)
            .build()

        val requestBuilder = original.newBuilder()
            .url(url)

        val request = requestBuilder.build()
        return chain.proceed(request)
    }
}
